diff --git a/include/psdk_inc/intrin-impl.h b/include/psdk_inc/intrin-impl.h
index 0267f93a2..0c9db2e6a 100644
--- a/include/psdk_inc/intrin-impl.h
+++ b/include/psdk_inc/intrin-impl.h
@@ -2016,7 +2016,7 @@ void __cpuid(int CPUInfo[4], int InfoType) {
 #define __INTRINSIC_DEFINED___cpuid
 #endif /* __INTRINSIC_PROLOG */
 
-#if (!defined(__GNUC__) || __GNUC__ < 11)
+#if (!defined(__GNUC__) || __GNUC__ < 11) && !defined(__clang__)
 #if __INTRINSIC_PROLOG(__cpuidex)
 void __cpuidex(int CPUInfo[4], int, int);
 #if !__has_builtin(__cpuidex)
